### PR TITLE
Improve packaging + fix test

### PR DESCRIPTION
**Pull request by jcarpent**
@cmastalli As in #645 you did not provide the write access on top of your commits, I open this PR which provides different fixes with recent versions of Boost mainly.
It also provides support for OSX systems.
More will come soon to speed up (really!) the compilation of Python bindings.

### CODE REVIEW

**Comment by VladimirIvan** [comment]
Does this PR build on another one? If so, please leave a comment to first pull that PR so that the changes are easier to spot/review.

File: unittest/bindings/test_force_cost.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 u = pinocchio.utils.rand(ACTUATION.nu)
 MODEL.calcDiff(DATA, x, u)
 MODEL_ND.calcDiff(dnum, x, u)
-assert (np.allclose(DATA.Fx, dnum.Fx, atol=1e5 * MODEL_ND.disturbance))
-assert (np.allclose(DATA.Fu, dnum.Fu, atol=1e5 * MODEL_ND.disturbance))
-assert (np.allclose(DATA.Lx, dnum.Lx, atol=1e5 * MODEL_ND.disturbance))
-assert (np.allclose(DATA.Lu, dnum.Lu, atol=1e5 * MODEL_ND.disturbance))
+assert (np.allclose(DATA.Fx, dnum.Fx, atol=3e5 * MODEL_ND.disturbance))
```
Comment (cmastalli) [inline unittest/bindings/test_force_cost.py]: Why do you increase it? It seems to work pretty well 1e5 in my machine

**Comment by cmastalli** [comment]
> Does this PR build on another one? If so, please leave a comment to first pull that PR so that the changes are easier to spot/review.

Yes, it does. I merged the previous PR, so now you can easily check the contributions of this particular one.

**Comment by cmastalli** [comment]
@jcarpent thanks for working on the mac osx support. Could let me know if you try to run the examples and benchmarks successfully?

File: unittest/bindings/test_force_cost.py (hunk 1 above)
Comment (cmastalli) [reply unittest/bindings/test_force_cost.py]: I see it

**Comment by cmastalli** [comment]
I will add the formatting in a follow-up PR

**Comment by jcarpent** [comment]
@cmastalli Everything works well. I just modified a bit of the tests to make them passed.
I don't have any issues with Pinocchio @2.2.1. So, the problem with your CI is certainly not related to Pinocchio.

**Pull request merged**